filecoin-virtual-machine/fvm-contract-overview.md

```diff
@@ -6,12 +6,112 @@ description: >-
+- The Base Interface: A user deploys deploys a smart contract that inherits from [`IAggregatorOracle.sol`](https://github.com/xBalbinus/fevm-data-segment/blob/main/contracts/aggregator-oracle/IAggregatorOracle.sol) which allows them to rely on their own contract deployment to submit and complete data deals.
+- The Full Interface: A user relies on an existing FVM contract deployment to submit and complete data deals. Usually, the deployment is verified and maintained by storage infrastructures such as Lighthouse.
 
-<table><thead><tr><th width="69">#</th><th width="196">Function Name</th><th width="174">Purpose</th><th width="155">Key Parameters</th><th>Outcome</th></tr></thead><tbody><tr><td>1</td><td><code>submit</code></td><td>Function that submits a new request to the oracle</td><td><code>_cid</code></td><td><code>Event: SubmitAggregatorRequest</code></td></tr><tr><td>2</td><td><code>getAllDeals</code></td><td>Get all deal IDs for a specified cid</td><td><code>_cid</code></td><td><code>Deal[]</code></td></tr><tr><td>3</td><td><code>getActiveDeals</code></td><td>return all the _cid's active dealIds</td><td><code>_cid</code></td><td><code>Deal[]</code></td></tr><tr><td>4</td><td><code>getExpiringDeals</code></td><td>return all the deals' dealIds if they are expiring within <code>epochs</code></td><td><code>_cid</code></td><td><code>Deal[]</code></td></tr></tbody></table>
+Among the two interfaces, some important common features stand out:
+
+<table><thead><tr><th width="69">#</th><th width="196">Function Name</th><th width="174">Purpose</th><th width="155">Key Parameters</th><th>Outcome</th></tr></thead><tbody><tr><td>1</td><td><code>submit</code></td><td>Function that submits a new deal request to the oracle. Critical in all RaaS functionalities.</td><td><code>_cid</code></td><td><code>Event: SubmitAggregatorRequest</code></td></tr><tr><td>2</td><td><code>getAllDeals</code></td><td>Get all deal IDs for a specified cid</td><td><code>_cid</code></td><td><code>Deal[]</code></td></tr><tr><td>3</td><td><code>getActiveDeals</code></td><td>return all the _cid's active dealIds. Critical for replication deals.</td><td><code>_cid</code></td><td><code>Deal[]</code></td></tr><tr><td>4</td><td><code>getExpiringDeals</code></td><td>return all the deals' dealIds if they are expiring within <code>epochs</code>. Critical for renewal and repair jobs.</td><td><code>_cid, epochs</code></td><td><code>Deal[]</code></td></tr></tbody></table>
+
+## Interacting with the Smart Contract
+
+First, you'll need to either:
+- Start an instance of the BaseInterface by deploying a contract that inherits from `IAggregatorOracle` (you can do so via. `yarn deploy` in the [RaaS Starter Kit](https://github.com/filecoin-project/raas-starter-kit))
+- Or, use an existing instance of the FullInterface located at 
+  - **Calibration Testnet**: `0x6ec8722e6543fB5976a547434c8644b51e24785b`
```

Comment:
Does this have contract verification already?

Reply:
Also, this should talk about how to call submit() to Lighthouse's implementation of DealStatus. You should describe  how this works in the context of Lighthouse -- mock file, but no RaaS workers

(Wil defer to you on the location of this information -- might not be this exact section? IDK)

 cc @nandit123 @ravish1729

Reply:
See more here: https://docs.google.com/document/d/1vFeIUx9sjw-DMB9goZwNAvrnzyiJJ6XYUMap5W15L8w/edit#bookmark=id.vqdta42yazy